pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -702,25 +706,45 @@ func (s *DiscoveryServer) startPush(req *model.PushRequest) {
 	}
 }
 
-func proxyNeedsPush(con *XdsConnection, targetNamespaces map[string]struct{}) bool {
+func ProxyNeedsPush(proxy *model.Proxy, targetNamespaces map[string]struct{}, configs map[string]struct{}) bool {
```

Comment:
I presume this is called before computing the configs?

Reply:
This is called after all init like PushContext generation, but right before XDS config generation